qiskit_dynamics/solvers/fixed_step_solvers.py

```diff
@@ -299,18 +302,139 @@ def jax_expm_parallel_solver(
     y0: Array,
     max_dt: float,
     t_eval: Optional[Union[Tuple, List, Array]] = None,
+    magnus_order: Optional[int] = 1,
 ):
-    """Parallel version of jax_expm_solver implemented with JAX parallel operations."""
+    """Parallel version of jax_expm_solver implemented with JAX parallel operations.
 
-    def take_step(generator, t, h):
-        eval_time = t + 0.5 * h
-        return jexpm(generator(eval_time) * h)
+        Args:
+        generator: Generator for the LMDE.
+        t_span: Interval to solve over.
+        y0: Initial state.
+        max_dt: Maximum step size.
+        t_eval: Optional list of time points at which to return the solution.
+        magnus_order: The expansion order in the Magnus method. Only orders 1, 2, and 3
+            are supported.
+
+    Returns:
+        OdeResult: Results object.
+
+    """
+
+    take_step = get_exponential_take_step(magnus_order, expm_func=jexpm, just_propagator=True)
 
     return fixed_step_lmde_solver_parallel_template_jax(
         take_step, generator=generator, t_span=t_span, y0=y0, max_dt=max_dt, t_eval=t_eval
     )
 
 
+def matrix_commutator(m1, m2):
+    """This function computes the commutator of two matrices.
+
+    Args:
+        m1, m2: Two equal-dimension square matrices.
+
+    Returns:
+        matrix commutator of m1 and m2
+    """
+
+    return m1 @ m2 - m2 @ m1
+
+
+def get_exponential_take_step(
+    magnus_order: int, expm_func: Callable, just_propagator: Optional[bool] = False
```

Comment:
my understanding is that Optional[T] is equivalent to Union[T, None], so that Optional is actually incorrect here as it should always be a bool.